src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FileRenameTracker.cs

```diff
@@ -45,39 +45,44 @@ public FileRenameTracker(IUnconfiguredProjectVsServices projectVsServices, Visua
             _roslynServices = roslynServices;
         }
 
-        public async Task HintedAsync(IImmutableDictionary<Guid, IImmutableSet<IProjectChangeHint>> hints)
+        public Task HintedAsync(IImmutableDictionary<Guid, IImmutableSet<IProjectChangeHint>> hints)
         {
-            var files = hints.GetValueOrDefault(ProjectChangeFileSystemEntityRenameHint.RenamedFile) ?? ImmutableHashSet.Create<IProjectChangeHint>();
+            var files = hints[ProjectChangeFileSystemEntityRenameHint.RenamedFile];
             if (files.Count == 1)
             {
-                var hint = files.First() as IProjectChangeFileRenameHint;
-                if (hint != null && !hint.ChangeAlreadyOccurred)
+                var hint = (IProjectChangeFileRenameHint)files.First();
```

Comment:
All file rename events are instances of IProjectChangeFileRenameHint, if they aren't - then its a bug in the product.